lxd/operations.go

```diff
@@ -153,17 +154,18 @@ func (op *operation) Cancel() (chan error, error) {
 		return nil, fmt.Errorf("Only running operations can be cancelled")
 	}
 
+	op.lock.Lock()
 	if !op.mayCancel() {
+		op.lock.Unlock()
```

Comment:
Before, the Lock() was performed later (see the negative change), so this branch was not run with lock.
The reason I moved it up is that mayCancel() now looks at the channel, and there's a chance for a race condition if it returns true (because the channel is not nil) but before it locks another goroutine unsets it (by calling Cancellable(false)).